AiProgrammerApp.ts

```diff
@@ -46,6 +107,61 @@ export class AiProgrammerApp extends App {
             blockBuilder: this.blockBuilder,
         };
     }
+    public async authorizationCallback(
+        token: IAuthData,
+        user: IUser,
+        read: IRead,
+        modify: IModify,
+        http: IHttp,
+        persistence: IPersistence
+    ) {
+        const deleteTokenTask = {
+            id: ProcessorsEnum.REMOVE_GITHUB_LOGIN,
+            when: "7 days",
+            data: {
+                user: user,
+                config: this.oauth2Config,
+            },
+        };
+        let text = `GitHub Authentication Succesfull 🚀`;
+        let interactionData = await getInteractionRoomData(
+            read.getPersistenceReader(),
+            user.id
+        );
+
+        if (token) {
+            await modify.getScheduler().scheduleOnce(deleteTokenTask);
+        } else {
+            text = `Authentication Failure 😔`;
+        }
+        if (interactionData && interactionData.roomId) {
+            let roomId = interactionData.roomId as string;
+            let room = (await read.getRoomReader().getById(roomId)) as IRoom;
+            await clearInteractionRoomData(persistence, user.id);
+            await sendNotification(read, modify, user, room, text);
+        } else {
+            await sendDirectMessage(read, modify, user, text, persistence);
+        }
+    }
+    public oauth2ClientInstance: IOAuth2Client;
+    public oauth2Config: IOAuth2ClientOptions = {
+        alias: "github-app",
+        accessTokenUri: "https://github.com/login/oauth/access_token",
+        authUri: "https://github.com/login/oauth/authorize",
+        refreshTokenUri: "https://github.com/login/oauth/access_token",
+        revokeTokenUri: `https://api.github.com/applications/client_id/token`,
+        authorizationCallback: this.authorizationCallback.bind(this),
+        defaultScopes: ["users", "repo", "gist"],
```

Comment:
repo scope should not be needed here.

Reply:
there's also a function to directly upload code pieces into Github repository, therefore still need this repo scope to ensure this function